src/query/encoding.ts

```diff
@@ -48,3 +48,38 @@ export function isMeasure(encQ: EncodingQuery) {
   return (encQ.type === Type.QUANTITATIVE && !encQ.bin) ||
       (encQ.type === Type.TEMPORAL && !encQ.timeUnit);
 }
+
+/**
+ *  Returns the true scale type of an encoding.
+ *  @returns {ScaleType} If the scale type was not specified, it is inferred from the encoding's Type.
+ *  @returns {undefined} If the scale type was not specified and Type (or TimeUnit if applicable) is an EnumSpec, there is no clear scale type
+ */
+
+export function scaleType(scaleType: ScaleType, timeUnit: TimeUnit | EnumSpec<TimeUnit> | ShortEnumSpec, type: Type | EnumSpec<Type> | ShortEnumSpec) {
+  if (scaleType !== undefined) {
+    return scaleType;
+  }
+
+  if (isEnumSpec(type)) {
+    return undefined;
+  }
+
+  if (type === Type.QUANTITATIVE) {
+    return ScaleType.LINEAR;
+  } else if (type === Type.ORDINAL || type === Type.NOMINAL) {
+    return ScaleType.ORDINAL;
+  } else if (type === Type.TEMPORAL) {
+    if (timeUnit !== undefined) {
+      if (isEnumSpec(timeUnit)) {
+        return undefined;
+      }
+      return defaultScaleType(timeUnit as TimeUnit);
+    } else {
+      return ScaleType.TIME;
+    }
+  } else {
+    /* istanbul ignore next */
```

Comment:
Is this truly the right location for ignore next?  Does test coverage say that this entire else branch is ignored?   (I doubt it should be one line above.)